tools/pipelines/test-perf-benchmarks.yml

```diff
@@ -258,17 +258,27 @@ stages:
               tar --extract --verbose --file $TAR_PATH ./src/test
 
         # Install package dependencies (this gets devDependencies installed so we can then run the package's tests)
-        # and run tests.
-        - task: Bash@3
-          displayName: Run memory performance tests - ${{ testPackage }}
+        # Note: the required .npmrc file is created by the include-test-perf-benchmarks.yml template above.
+        - task: CopyFiles@2
+          displayName: Copy .npmrc to test package folder - ${{ testPackage }}
           inputs:
-            targetType: 'inline'
-            workingDirectory: ${{ variables.testWorkspace }}/node_modules/${{ testPackage }}
-            script: |
-              echo "Run memory performance test for ${{ testPackage }}"
-              cp ${{ variables.testWorkspace }}/.npmrc . ;
-              npm i ;
-              npm run test:memory-profiling:report;
+            sourceFolder: ${{ variables.testWorkspace }}
+            contents: '.npmrc'
+            targetFolder: ${{ variables.testWorkspace }}/node_modules/${{ testPackage }}
+        - task: Npm@1
+          displayName: Install dependencies - ${{ testPackage }}
+          retryCountOnTaskFailure: 1
+          inputs:
+            command: 'install'
+            workingDir: ${{ variables.testWorkspace }}/node_modules/${{ testPackage }}
```

Comment:
Would a pnpm install get more reliability?

Reply:
Since this is that ugly place (or one of) in our infrastructure where we're installing from within a package's folder _after we install the package_, meaning no lockfiles or similar are in place, I'm not sure I'd expect much difference? Or if you mean because of the pnpm global store potentially being available locally thanks to cache, I don't _think_ the test pipelines currently cache it like the build pipelines do. If they were, then yeah.

Reply:
Not so much caching. I think pnpm is better at negotiating network issues and such.

Reply:
Caching can be added to any pipeline if that is useful.